mirdata/datasets/tonas.py

```diff
@@ -0,0 +1,409 @@
+
+
+def load_audio(fhandle: str) -> Tuple[np.ndarray, float]:
+    """Load a TONAS audio file.
+
+    Args:
+        fhandle (str): path to an audio file
+
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+
+    """
+    return librosa.load(fhandle, sr=44100, mono=True)
+
+
+@io.coerce_to_string_io
+def load_f0(fhandle: TextIO) -> Optional[F0Data]:
+    """Load TONAS f0 annotations
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to f0 annotation file
+
+    Returns:
+        F0Data: predominant f0 melody
+
+    """
+    times = []
+    freqs = []
+    freqs_corr = []
+    energies = []
+    reader = np.genfromtxt(fhandle)
+    for line in reader:
+        times.append(float(line[0]))
+        energies.append(float(line[1]))
+        freqs.append(float(line[2]))
+        freqs_corr.append(float(line[3]))
+
+    times = np.array(times, dtype="float")  # type: ignore
+    freqs = np.array(freqs, dtype="float")  # type: ignore
+    freqs_corr = np.array(freqs_corr, dtype="float")  # type: ignore
+    energies = np.array(energies, dtype="float")  # type: ignore
+    confidence = (cast(np.ndarray, freqs_corr) > 0).astype(float)
+
+    return F0Data(times, freqs, freqs_corr, energies, confidence)
+
+
+@io.coerce_to_string_io
+def load_notes(fhandle: TextIO) -> Optional[NoteData]:
+    """Load TONAS note data from the annotation files
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to a notes annotation file
+
+    Returns:
+        NoteData: note annotations
+
+    """
+    intervals = []
+    pitches = []
+    energy = []
+    confidence = []
+
+    reader = csv.reader(fhandle, delimiter=",")
+    tuning = next(reader)[0]
+    for line in reader:
+        intervals.append([line[0], float(line[0]) + float(line[1])])
+        # Convert midi value to frequency
+        note_hz = midi_to_hz(float(line[2]), float(tuning))
+        pitches.append(note_hz)
+        energy.append(float(line[3]))
+        confidence.append(1.0)
+
+    note_data = NoteData(
+        np.array(intervals, dtype="float"),
+        np.array(pitches, dtype="float"),
+        np.array(energy, dtype="float"),
+        np.array(confidence, dtype="float"),
+    )
+
+    return note_data
+
+
+@io.coerce_to_string_io
+def load_tuning_frequency(fhandle: TextIO) -> float:
+    """Load tuning frequency of the track with re
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to a notes annotation file
+
+    Returns:
+        tuning_frequency (float): returns new tuning frequency considering the deviation
+
+    """
+
+    # Compute tuning frequency
+    cents_deviation = float(next(csv.reader(fhandle, delimiter=","))[0])
+    tuning_frequency = 440 * (
+        2 ** (cents_deviation / 1200)
+    )  # Frequency of A (common value is 440Hz)
+
+    return tuning_frequency
+
+
+def midi_to_hz(midi_note, tuning_deviation):
```

Comment:
Same question as above + would this be equivalent to `librosa`'s [`midi_to_hz`](http://man.hubwiz.com/docset/LibROSA.docset/Contents/Resources/Documents/generated/librosa.core.midi_to_hz.html)?

Reply:
hmmm, the thing is, I wrote my function here to easily support the fact that the tunning freq is different from 440, and the info we get from TONAS about this new tunning freq is the deviation is cents from 440. I wrote the function to adapt to this data organization!